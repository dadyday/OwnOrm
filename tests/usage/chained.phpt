<?php
require_once 'cfg.php';

use OwnOrm\Model;
use Tester\Assert as Is;

// single calls usage w/o chain and magic

$oModel = Model::create()
	->entity('user')
		->property('name')
			->type('string')
			->length(30)
			->default('guest')
			->null(false)
			->unique()
		->property('password', 'string', null)
	->entity('login')
		->property('user_id', 'related:user')
		->property('logged', 'date', 'now')
		->index('userlogin', ['user_id', 'logged'])
;
#dump($oModel->entities);
include 'check.php';
