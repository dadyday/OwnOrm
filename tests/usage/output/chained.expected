OwnOrm\Entity::__set_state(/* #8ece */ array(
	'oModel' => OwnOrm\ModelFassade::__set_state(/* #2839 */ array(
		'_entity' => OwnOrm\Entity::__set_state(/* #29df */ array(
			'oModel' => /* OwnOrm\ModelFassade dumped on line 2 */,
			'name' => 'login',
			'aProperty' => array(
				'user_id' => OwnOrm\Property::__set_state(/* #64a7 */ array(
					'oEntity' => /* OwnOrm\Entity dumped on line 3 */,
					'name' => 'user_id',
					'null' => TRUE,
					'type' => NULL,
					'length' => 0,
					'default' => NULL,
					'related' => 'user',
				)),
				'logged' => OwnOrm\Property::__set_state(/* #f8b6 */ array(
					'oEntity' => /* OwnOrm\Entity dumped on line 3 */,
					'name' => 'logged',
					'null' => TRUE,
					'type' => 'string',
					'length' => 40,
					'default' => 'now',
					'related' => NULL,
				)),
			),
			'aIndex' => array(
				'userlogin' => OwnOrm\Index::__set_state(/* #a858 */ array(
					'oEntity' => /* OwnOrm\Entity dumped on line 3 */,
					'name' => 'userlogin',
					'type' => 'index',
					'aField' => array(
						'user_id' => /* OwnOrm\Property dumped on line 7 */,
						'logged' => /* OwnOrm\Property dumped on line 16 */,
					),
				)),
			),
		)),
		'_property' => /* OwnOrm\Property dumped on line 16 */,
		'_index' => /* OwnOrm\Index dumped on line 27 */,
		'_row' => NULL,
		'aEntity' => array(
			'user' => /* OwnOrm\Entity dumped on line 1 */,
			'login' => /* OwnOrm\Entity dumped on line 3 */,
		),
		'aRow' => array(),
	)),
	'name' => 'user',
	'aProperty' => array(
		'name' => OwnOrm\Property::__set_state(/* #c539 */ array(
			'oEntity' => /* OwnOrm\Entity dumped on line 1 */,
			'name' => 'name',
			'null' => FALSE,
			'type' => 'string',
			'length' => 30,
			'default' => 'guest',
			'related' => NULL,
		)),
		'password' => OwnOrm\Property::__set_state(/* #9d17 */ array(
			'oEntity' => /* OwnOrm\Entity dumped on line 1 */,
			'name' => 'password',
			'null' => TRUE,
			'type' => 'string',
			'length' => 40,
			'default' => NULL,
			'related' => NULL,
		)),
	),
	'aIndex' => array(
		'name' => OwnOrm\Index::__set_state(/* #dd44 */ array(
			'oEntity' => /* OwnOrm\Entity dumped on line 1 */,
			'name' => 'name',
			'type' => 'unique',
			'aField' => array('name' => /* OwnOrm\Property dumped on line 49 */),
		)),
	),
))
